lib/endpoints/class-wp-rest-posts-controller.php

```diff
@@ -341,16 +345,12 @@ public function delete_item( $request ) {
 			$result = wp_trash_post( $id );
 		}
 
-		if ( ! $result ) {
+		if ( $result ) {
```

Comment:
I've always thought it to be a bad pattern to check for a negative.

@joehoyle @rachelbaker ?